tools/NTP/plot_ntp_stats.py

```diff
@@ -0,0 +1,254 @@
+    "receive",                # The receive timestamp in seconds, Float
+    "transmit",               # The transmit timestamp in seconds, Float
+    "final",                  # The final timestamp in seconds, Float
+]
+
+# Sysstats present the NTP system statistics.
+# Each hour the NTP system outputs a line containing the following:
+SYSSTATS_KEYS = [
+    "modified_julian_day",    # MJD = Julian Day - 2400000.5 = Days since May 23, 1968, Int
+    "seconds_past_midnight",  # How many seconds have elapsed since UTC midnight? Float
+    "time_since_restart",     # The number of hours elapsed since system last rebooted, Int
+    "packets_received",       # The number of packets received, Int
+    "packets_processed",      # The number of packets received in response to previous packets sent, Int
+    "current_version",        # The number of packets received that matched the current version of NTP, Int
+    "previous_version",       # The number of packets received that matched the previous version of NTP, Int
+    "bad_version",            # The number of packets received that matched neither current nor prev version of NTP, Int
+    "access_denied",          # The number of packets denied access for any reason, Int
+    "bad_length",             # The number of packets with an invalid length, format or port number, Int
+    "bad_authentication",     # The number of packets not verified as authentic, Int
+    "rate_exceeded",          # The number of packets discarded due to rate limitation, Int
+]
+
+#PEER_STATUS_WORD = {
+#    "status",   b'0000000000011111'     #
+#    "select",   b'0000000011100000'     #
+#    "count",    b'0000111100000000'     #
+#    "code",     b'1111000000000000'     #
+#}
+
+
+# class NTPUtils(object):
+# TODO: place static methods into an NTPUtils class
+
+if __name__ == "__main__":
+    # TODO: Cleanup arguments to allow a list of plot types, or to plot all from available files
+    parser = ap.ArgumentParser(description='Analyze and plot NTP statistics')
+    parser.add_argument('--allan-deviation', help='Enable plotting Allan deviation',
+                        action='store_true')
+    parser.add_argument('--offset', help='Plot the loopstats offset value (s)',
+                        action='store_true')
+    parser.add_argument('--freq-offset', help='Plot the loopstats frequency offset value (ppm)',
+                        action='store_true')
+    parser.add_argument('--peer-offset', help='Plot the peerstats estimated offset from true time (s)',
+                        action='store_true')
+    parser.add_argument('--peer-delay', help='Plot the peerstats packet delay over the network (s)',
+                        action='store_true')
+    parser.add_argument('--peer-dispersion', help='Plot the peerstats estimated dispersion (s)',
+                        action='store_true')
+    parser.add_argument('--ntp-config', help='Where is the NTP config file located?',
```

Comment:
so I don't know the name of the process that uses that ntp config file but you may want to mention that the ntp config file that should be specified here is the one loaded by the process right?. (ntpd??)

Reply:
also could use a custom usage message for this script if you like.